### PR TITLE
[#IP-177] Imported @pagopa/io-functions-commons

### PR DESCRIPTION
**Pull request by michaeldisaro**
#### List of Changes
- Imported @pagopa/io-functions-commons
- Updated some cross dependencies (fp-ts)
- Replaced all old imports

#### Motivation and Context
We want to clean io-functions-admin to be in a coherent state with io-functions-commons.

#### How Has This Been Tested?
It has been tested by performing:
- yarn install --frozen-lockfile
- yarn build

BUILD FAILS WITH COMPILATION ERRORS. Errors are spread through various packages from commons to azure.

#### Types of changes

- [x] Chore (nothing changes by a user perspective)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by balanza** [comment]
Changes introduced in `SetUserDataProcessingStatusActivity/handler.ts` weren't enough in order to handle new typings for `UserDataProcessing` model. That highlighted we introduced unwanted complexity in https://github.com/pagopa/io-functions-commons/pull/142 and now we have to deal with it.

In https://github.com/pagopa/io-functions-admin/pull/144/commits/718ddecc9f25c9620bd8367f9117d70e89695fa0 I tried to  solve it

File: package.json [hunk 1]
```diff
@@ -74,9 +75,8 @@
     "dotenv": "^8.2.0",
     "durable-functions": "^1.4.3",
     "express": "^4.15.3",
-    "fp-ts": "1.17.0",
+    "fp-ts": "1.17.4",
     "html-to-text": "^5.1.1",
-    "io-functions-commons": "^17.0.0",
     "io-functions-express": "^1.1.0",
     "io-ts": "1.8.5",
     "italia-ts-commons": "^8.5.0",
```
Comment (BurnedMarshal) [inline package.json:82]: Can we remove this package to use only `@pagopa/ts-commons`? Or we are planning this upgrade into another story/PR?

Comment (balanza) [reply package.json:82]: Intentionally yes, let me see if we can in this same PR
@michaeldisaro thoughts?

Comment (michaeldisaro) [reply package.json:82]: I wanted to do that too, but in another PR. If you agree we can manage to remove it in this PR. Anyway it's a good thing to get rid of the old dependency.

**Comment by michaeldisaro** [comment]
I converted this to draft to wait for the new io-functions-commons package because the last commit is developed to use it. I will request the review when https://github.com/pagopa/io-functions-commons/pull/162 will be merged and release, and I will bump the package version here.

**Review by balanza — APPROVED** [review]

**Pull request merged**